pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -16,10 +17,12 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 )
 
 const (
 	BundleLookupConditionPacked v1alpha1.BundleLookupConditionType = "BundleLookupNotPersisted"
+	exclusionAnnotation         string                             = "olm.operatorframework.io/exclude-global-namespace-resolution"
```

Comment:
So, this is now part of the "API"? Is this documented?

Reply:
It's coming from [this PR that added the feature](https://github.com/operator-framework/operator-lifecycle-manager/pull/2788)  (see [ref](https://github.com/operator-framework/operator-lifecycle-manager/pull/2788/files#diff-11677c62612a788aa62feb63e14540b299fbc9dcd8b35ca7e3576b5fd6058e82R29)). 

So it's an existing feature, I'm just moving the usage to a different area in this PR. However, now that you bring it up, fyi I also have an experiment going on [here](https://github.com/operator-framework/operator-lifecycle-manager/pull/3360) to essentially redo the PR that added this feature, using only the change I made in this PR.

Reply:
Avoided removing the `og-source-provider` in this PR because it needs to be back ported and wanted to keep the scope small. But I think that makes a lot of sense as a follow up immediately.